prep-exercises/h_enums.py

```diff
@@ -0,0 +1,126 @@
+    MACOS = "macOS"
+    ARCH = "Arch Linux"
+    UBUNTU = "Ubuntu"
+
+@dataclass(frozen=True)
+class Person:
+    name: str
+    age: int
+    preferred_operating_systems: OperatingSystem
+
+
+@dataclass(frozen=True)
+class Laptop:
+    id: int
+    manufacturer: str
+    model: str
+    screen_size_in_inches: float
+    operating_system: OperatingSystem
+
+def find_possible_laptops(available_laptops: List[Laptop], current_person: Person) -> List[Laptop]:
+    return [
+        laptop for laptop in available_laptops
+        if laptop.operating_system == current_person.preferred_operating_systems
+    ]
+
+laptops = [
+    Laptop(id=1, manufacturer="Dell", model="XPS", screen_size_in_inches=13, operating_system=OperatingSystem.ARCH),
+    Laptop(id=2, manufacturer="Dell", model="XPS", screen_size_in_inches=15, operating_system=OperatingSystem.UBUNTU),
+    Laptop(id=3, manufacturer="Dell", model="XPS", screen_size_in_inches=15, operating_system=OperatingSystem.UBUNTU),
+    Laptop(id=4, manufacturer="Apple", model="MacBook", screen_size_in_inches=13, operating_system=OperatingSystem.MACOS),
+    Laptop(id=5, manufacturer="Apple", model="MacBook Air", screen_size_in_inches=13, operating_system=OperatingSystem.MACOS),
+    Laptop(id=6, manufacturer="Lenovo", model="ThinkPad", screen_size_in_inches=14, operating_system=OperatingSystem.ARCH),
+    Laptop(id=7, manufacturer="Asus", model="ZenBook", screen_size_in_inches=13, operating_system=OperatingSystem.UBUNTU),
+    Laptop(id=8, manufacturer="HP", model="Spectre", screen_size_in_inches=14, operating_system=OperatingSystem.MACOS),
+    Laptop(id=9, manufacturer="Apple", model="MacBook Pro", screen_size_in_inches=16, operating_system=OperatingSystem.MACOS),
+]
+
+# ------------------------
+# Exercise 1
+# ------------------------
+#  Write a program which:
+#  - Already has a list of Laptops that a library has to lend out.
+#  - Accepts user input to create a new Person - it should use the input function to read a person’s name, age, 
+#    and preferred operating system.
+#  - Tells the user how many laptops the library has that have that operating system.
+#  - If there is an operating system that has more laptops available, tells the user that if they’re willing 
+#    to accept that operating system they’re more likely to get a laptop.
+#  You should convert the age and preferred operating system input from the user into more constrained types as quickly 
+#  as possible, and should output errors to stderr and terminate the program with a non-zero exit code if the
+#  user input bad values.
+
+def user_prompt() -> Person:
+    try:
+        name = (input("Please enter your first name: ")).strip()
+        if not name.isalpha():
+            raise ValueError("Name must contain only alphabetic characters.")
+        
+        age = int(input("Please enter your age: "))
+        minimum_age = 18
+        if age < minimum_age: 
+            raise ValueError("Age must be 18 or over.")
```

Comment:
Would be better to embed minimum_age into the error message, just in case it changes.